### DISCUSSION TITLE
Custom plugin extending Stripe/user functionality

### DISCUSSION THREAD

**Discussion by D-Un1t** [impress-org/givewp#6150] [Developer Q&A] [answered]
Hi,
For a new project I'd like to use Stripe and recurring payments, and users with accounts only. I would like the user to be able to save  their credit card information (only a token locally, the actual card on Stripe), so they can support different causes without having to reenter their card info all the time. Ideally, I'd have a user enter their card details right away upon registration.

In order to realize this, I'm looking at writing a custom plugin. I would be looking at adding a Stripe PaymentMethod to every WP/GiveWP account created, displaying data from this PaymentMethod to the user and obviously using this data for 'purchases'. 

I was wondering if the GiveWP and Stripe gateway plugins are coded in such a way that an extra plugin hooking into certain actions/filters could be written on top of it, adding these specific features? Or would I really be forced to alter the plugin code itself? 

Thanks for any insights!

**Comment by JasonTheAdams** [comment, marked as answer]
Hi @daancjanssen!

You should have no problem writing an additional plugin which uses the hooks provided in both. Modifying the plugin code is not advised and should not be necessary. If, in the process of building your plugin, you realize you're unable to do something because we don't have an action or filter in place to make it possible, please start a new discussion to let us know about the specific hook that would help you!

**Reply by D-Un1t** [reply]
Good to hear, thanks @JasonTheAdams!
